scala3/new-in-scala3.md

```diff
@@ -0,0 +1,112 @@
+of the type-system have been change to be more principled. While this also
+brings exciting new features along (like union types), first and foremost, it
+means that the type-system gets (even) less in your way and for instance
+[type-inference][type-inference] and overload resolution are much improved.
+
+### New & Shiny: The Syntax
+Besides many (minor) cleanups, the Scala 3 syntax offers the following  improvements:
+
+- A new "quiet" syntax for control structures like `if`, `while`, and `for` ([new control syntax][syntax-control])
+- The `new` keyword is optional (_aka_ [creator applications][creator])
+- [Optional braces][syntax-indentation] that supports a distraction-free, indentation sensitive style of programming
+- Change of [type-level wildcards][syntax-wildcard] from `_` to `?`.
+- Implicits (and their syntax) have been [heavily revised][implicits].
+
+### Opinionated: Contextual Abstractions
+One underlying core concept of Scala was (and still is to some degree) to
+provide users with a small set of powerful features that can be combined to
+great (and sometimes even unforeseen) expressivity. For example, the feature of _implicits_
+has been used to model contextual abstraction, to express type-level
+computation, model type-classes, perform implicit coercions, encode
+extension methods, and many more.
+Learning from these use cases, Scala 3 takes a slightly different approach
+and focuses on **intent** rather than **mechanism**.
+Instead of offering one very powerful feature, Scala 3 offers multiple
+tailored language features, allowing programmers to directly express their intent:
+
+- **Abtracting over contextual information**. [Using clauses][contextual-using] allow programmers to abstract over information that is available in the calling context and should be passed implicitly. As an improvement over Scala 2 implicits, using clauses can be specified by type, freeing function signatures from term variable names that are never explicitly referred to.
+
+- **Providing Type-class instances**. [Given instances][contextual-givens] allow programmers to define the _canonical value_ of a certain type. This makes programming with type-classes more straightforward without leaking implementation details.
+
+- **Retroactively extending classes**. In Scala 2, extension methods had to be encoded using implicit conversions or implicit classes. In contrast, in Scala 3 [extension methods][contextual-extension] are now directly built into the language, leading to better error messages and improved type inference.
+
+- **Viewing one type as another**. Implicit conversion have been [redesigned][contextual-conversions] from the ground up as instances of a type-class `Conversion`.
+
+- **Higher-order contextual abstractions**. The _all-new_ feature of [context functions][contextual-functions] makes contextual abstractions a first-class citizen. They are an important tool for library authors and allow to express concise domain specific languages.
```

Comment:
I would like to also advertise “import suggestions” (https://www.scala-lang.org/blog/2020/05/05/scala-3-import-suggestions.html), I believe they will make it dramatically easier to deal with implicit parameters.

Reply:
Could you write one sentence on this -- maybe you have a good selling point? I am happy to revise and add a link.

Reply:
That’s a good blog post, I’m sorry that I didn’t know about it. I just made a note to reference it from the given/using pages as well.

Reply:
What about the following?

> - **Actionable feedback from the compiler**. In case an implicit parameter could not be resolved by the compiler, it now provides you [import suggestions](https://www.scala-lang.org/blog/2020/05/05/scala-3-import-suggestions.html) that may fix the problem.

Reply:
Thanks, I added it to the list!

Reply:
That sounds good to me. I’ll be glad to add it today.

Reply:
Oops, I see now that Jonathan took care of that. Thanks!